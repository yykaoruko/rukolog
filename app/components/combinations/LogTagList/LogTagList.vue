<template>
  <ul class="log-tag-list">
    <template v-for="(item, index) in tags">
      <li :key="index" class="log-tag-list__item">
        <app-label :label="item" />
      </li>
    </template>
  </ul>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue';
import AppLabel from '@/components/basics/AppLabel/AppLabel.vue';
export default Vue.extend({
  name: 'LogTagList',
  components: {
    AppLabel,
  },
  props: {
    tags: {
      type: Array as PropType<string[]>,
      required: true,
    },
  },
});
</script>

<style lang="scss" scoped>
.log-tag-list {
  color: inherit;
}
</style>
