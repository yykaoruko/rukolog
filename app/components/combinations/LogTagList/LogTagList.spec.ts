import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import LogTagList from './LogTagList.vue';

describe('LogTagList', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LogTagList, {
        propsData: {
          tags: ['hoge', 'huga'],
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
