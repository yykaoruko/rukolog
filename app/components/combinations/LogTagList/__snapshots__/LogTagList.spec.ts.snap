// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogTagList mount snapshot 1`] = `
<ul
  class="log-tag-list"
>
  <li
    class="log-tag-list__item"
  >
    <app-label-stub
      label="hoge"
    />
  </li>
  <li
    class="log-tag-list__item"
  >
    <app-label-stub
      label="huga"
    />
  </li>
</ul>
`;
